tests/unit/s2n_sslv3_test.c

```diff
@@ -0,0 +1,133 @@
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+
+#define S2N_TEST_DATA_SIZE 100
+
+S2N_RESULT s2n_test_send_receive_data(struct s2n_connection *sender, struct s2n_connection *receiver)
+{
+    uint8_t test_data[S2N_TEST_DATA_SIZE] = { 0 };
+    struct s2n_blob test_data_blob = { 0 };
+    EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
+    EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
+
+    /* Send data */
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    ssize_t bytes_written = 0;
+    while (bytes_written < S2N_TEST_DATA_SIZE) {
+        ssize_t w = s2n_send(sender, test_data + bytes_written, S2N_TEST_DATA_SIZE - bytes_written, &blocked);
+        EXPECT_TRUE(w >= 0);
+        bytes_written += w;
+    }
+
+    /* Receive data */
+    uint8_t buffer[S2N_TEST_DATA_SIZE] = { 0 };
+    ssize_t bytes_received = 0;
+    while (bytes_received < S2N_TEST_DATA_SIZE) {
+        ssize_t r = s2n_recv(receiver, buffer + bytes_received, S2N_TEST_DATA_SIZE - bytes_received, &blocked);
+        EXPECT_TRUE(r > 0);
+        bytes_received += r;
+    }
+
+    EXPECT_BYTEARRAY_EQUAL(test_data, buffer, S2N_TEST_DATA_SIZE);
+
+    return S2N_RESULT_OK;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Ensure that AWS-LC supports the SSLv3 HMAC. */
+    if (s2n_libcrypto_is_awslc()) {
+        EXPECT_TRUE(s2n_hmac_is_available(S2N_HMAC_SSLv3_MD5));
+    }
+
+    /* Skip the tests if SSLv3 isn't supported. */
+    if (!s2n_hmac_is_available(S2N_HMAC_SSLv3_MD5)) {
+        END_TEST();
+    }
+
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *rsa_chain_and_key = NULL, s2n_cert_chain_and_key_ptr_free);
+    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&rsa_chain_and_key,
+            S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *ecdsa_chain_and_key = NULL, s2n_cert_chain_and_key_ptr_free);
+    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&ecdsa_chain_and_key,
+            S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN, S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+    char dhparams_pem[S2N_MAX_TEST_PEM_SIZE] = { 0 };
+    EXPECT_SUCCESS(s2n_read_test_pem(S2N_DEFAULT_TEST_DHPARAMS, dhparams_pem, S2N_MAX_TEST_PEM_SIZE));
+
+    /* Self-talk test */
+    for (size_t i = 0; i < security_policy_test_all.cipher_preferences->count; i++) {
+        struct s2n_cipher_suite *cipher_suite = security_policy_test_all.cipher_preferences->suites[i];
+
+        /* Skip non-sslv3 cipher suites */
+        if (!cipher_suite->sslv3_record_alg) {
+            continue;
+        }
+
+        /* Skip unsupported record algorithms */
+        if (!cipher_suite->sslv3_record_alg->cipher->is_available()) {
+            continue;
+        }
```

Comment:
It might be worth making sure that this test triggers / doesn't just `continue` at least once so we're sure that we're really testing sslv3 somewhere.